languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -17896,35 +17950,77 @@ though, and has since been slightly modified)-->
     </rule>
   </category>
   <category id="CAT_ANGLICISMES" name="Anglicismes (calques, emprunts directs, etc.)" type="style">
-      <rule id="TEMPLATE_ANGLICISME" name="Anglicisme : template" default="temp_off">
+      <rulegroup id="SCREENSHOT" name="Anglicisme : screenshot" default="temp_off">
+          <rule>
+              <pattern>
+                  <token regexp="yes">screenshots?</token>
+              </pattern>
+              <message>« \1 » est un anglicisme. L'équivalent français est « capture(s) d'écran ».</message>
+              <suggestion><match no="1" regexp_match="(screenshot)(s)?" regexp_replace="capture$2"/> d'écran</suggestion>
+              <example correction="capture d'écran">J'en ai pris un <marker>screenshot</marker> si tu veux.</example>
+              <example correction="captures d'écran">J'en ai pris des <marker>screenshots</marker> si tu veux.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token>screen</token>
+                  <token regexp="yes">shots?</token>
+              </pattern>
+              <message>« \1 \2 » est un anglicisme. L'équivalent français est « capture(s) d'écran ».</message>
+              <suggestion><match no="2" regexp_match="(shot)(s)?" regexp_replace="capture$2"/> d'écran</suggestion>
+              <example correction="capture d'écran">J'en ai pris un <marker>screen shot</marker> si tu veux.</example>
+              <example correction="captures d'écran">J'en ai pris un <marker>screen shots</marker> si tu veux.</example>
+          </rule>
+      </rulegroup>
+      <rulegroup id="SCREENCAST" name="Anglicisme : screencast" default="temp_off">
+          <rule>
+              <pattern>
+                  <token regexp="yes">screencasts?</token>
+              </pattern>
+              <message>« \1 » est un anglicisme. L'équivalent français est « capture(s) vidéo ».</message>
+              <suggestion><match no="1" regexp_match="(screencast)(s)?" regexp_replace="capture$2"/> vidéo</suggestion>
```

Comment:
Will this also work when "Screencast" is at the beginning of a sentence and therefore capitalized? @vkyfox

Reply:
Same applies to the other rules.

Reply:
@vkyfox You need to add `regexp_match="(?iu)(screencast)(s)?"` to make it case-insensitive.

Reply:
Such does not seem to be the case in <pattern> rules as the capitalized examples I added are detected in the test runs. I suppose this has to do with the case_sensitive attribute for the token taking priority on the regexp itself?
I had to adjust the regexp in the suggestion for it to work properly, however, so thank you for bringing this to my attention!

Reply:
The pattern rules are case-insensitive by default. But the `regexp_match` (for generating suggestions) is case-sensitive. You can do `regexp_match="([Ss]creenshot)(s)?"` (for capitalized words) or regexp_match="(?iu)(screencast)(s)?" (fully case-insensitive).

Reply:
I did.